frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -114,7 +114,24 @@
 //! good solution is queued, then the fallback strategy [`pallet::Config::Fallback`] is used to
 //! determine what needs to be done. The on-chain election is slow, and contains no balancing or
 //! reduction post-processing. See [`onchain::OnChainSequentialPhragmen`]. The
-//! [`FallbackStrategy::Nothing`] should probably only be used for testing, and returns an error.
+//! [`FallbackStrategy::Nothing`] just returns an error, and enables the [`Phase::Emergency`].
+//!
+//! ### Emergency Phase
+//!
+//! If, for any of the below reasons:
+//!
+//! 1. No signed and unsigned solution submitted
+//! 2. Internal error
+//! 3. Fallback being `None`
+//!
+//! A call to `T::ElectionProvider::elect` is made, and `Ok(_)` cannot be returned, then the pallet
+//! proceeds to the [`Phase::Emergency`]. During this phase, any solution can be submitted from
+//! [`T::ForceOrigin`], without any checking. Once submitted, the forced solution is kept in
```

Comment:
> any solution can be submitted from [`T::ForceOrigin`], without any checking.

Why no checking? Could this make us more vulnerable due to human error?

IIUC Council/TC would submit a solution via `ForceOrigin` with a refferenda (please correct me if I am wrong). What if the individual who made the refferenda introduced a typo into the solution? Could this lead some a panic etc?

Reply:
It seems unlikely to me that a human will be typing out the solution. More likely they'll be uploading some data file generated by the solution miner or equivalent.

Reply:
From my (limited) experience, external application input, regardless of how it is generated, should always have some sort of check i.e. shit happens

In this case, it doesn't make sense to use `feasibility_check` because we may want to intentionally ignore some of checks (e.g. min score, desired_targets, etc). But what if just checked things like can we deserialize the solution, are the winners sane etc; basically just is the set sane and usable.

Maybe a better question: what happens if there is a fatal error with the solution?

Reply:
The solution is decoded into the type: `ReadySolution`. But no other check are made.

I think the check should be made offchain depending on the ForceOrigin implementation. If ForceOrigin is root for instance then the voter in democracy should check themself before approving it.

Reply:
> If ForceOrigin is root for instance then the voter in democracy should check themself before approving it.

If we expect offchain checking, would be good to have an example of how to do it 

So for something like this would you imagine a little cli that reads in the serialized data for `ReadySolution` and then checks
- validator IDs are actual validators
- nominators are actual nominators
?

(sorry for all the pestering questions, just want to make sure I understand 😬)

Reply:
@emostov the staking miner will have a simple subcommand to generate this for emergencies. And your assumption is right. `ForceOrigin` must be trusted and the solution should not be checked here, otherwise we are at risk of all sorts of other issues.

Reply:
@kianenigma shouldn't this PR be blocked then until we have the staking miner?

And shouldn't voters be able to verify the staking miner?